### PR TITLE
Sign In, Sign Up and Forgot Password UI

### PR DESCRIPTION
**Pull request by shekokar**
1. Updated UI of Sign Up page

Screenshot from 2020-12-29 11-25-10

2. Updated UI of Sign In page (Connection and backend route for  'Remember Me' needed)

Screenshot from 2020-12-29 11-25-14

3. Created UI for reset password page (connection and backend routes needed) - Front-end link: localhost:3000/reset

Screenshot from 2020-12-29 11-25-18
    
4. Created UI for new password page (connection and backend routes needed) - Front-end link: localhost:3000/newpass/:token

Screenshot from 2020-12-29 11-25-58

### CODE REVIEW

File: new_client/src/components/auth/Signin.jsx [hunk 1]
```diff
@@ -48,7 +100,8 @@ function Signin(props) {
 	function handleClick(event) {
 		event.preventDefault();
 		if (isFormValid()) {
-			const user = { username, password };
+      const user = { username, password };
+      alert(`${username},${password}`)
```
Comment (Saif807380) [inline new_client/src/components/auth/Signin.jsx:104]: This alert needs to be removed.

**Comment by Saif807380** [comment]
Thanks for the PR and Great Work! Just one change - 
- The links to `forgot password` and `sign up` should be in green on the `sign in` page as they are in green on the `sign up` page.

File: new_client/src/components/auth/Signup.jsx [hunk 2]
```diff
@@ -150,85 +203,100 @@ function Signup(props) {
...
+				color='#52b107'
+				required
+				id="email"
+				label="Email Address"
+				name="email"
+				autoComplete="email"
+				style={{color:'#52b107',borderColor:'#52b107'}}
+				onChange={handleChangeEmail}
+		  autoFocus
+		  InputProps = {{startAdornment: <InputAdornment position="start"><AlternateEmailIcon/></InputAdornment>}}
```
Comment (ShubhankarKG) [inline new_client/src/components/auth/Signup.jsx:245]: Can we use `<EmailIcon />` here? That one makes more sense in my opinion.

File: new_client/src/components/auth/Signup.jsx [hunk 3]
```diff
@@ -150,85 +203,100 @@ function Signup(props) {
...
+				className={classes.submit}
+				size='large'
+				onClick={handleClick}
+				style={{backgroundColor:'#52b107'}}>
+				Sign Up
+			  </Button>
+				<Grid container item>
+				  <Typography>
+				  <Link style={{color:'#52b107',fontSize:15}} to="/signin">
+					{"Have Account? Sign In"}
```
Comment (ShubhankarKG) [inline new_client/src/components/auth/Signup.jsx:288]: nit: Have an account?

File: new_client/src/components/auth/ResetPw.jsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            fullWidth
+            color='#52b107'
+            required
+            label="Email ID"
+            placeholder="Enter registered Email ID"
+            name="username"
+            style={{color:'#52b107',borderColor:'#52b107'}}
+            onChange={handleEmail}
+      autoFocus
+      InputProps = {{startAdornment: <InputAdornment position="start"><AlternateEmailIcon/></InputAdornment>}}
```
Comment (ShubhankarKG) [inline new_client/src/components/auth/ResetPw.jsx:110]: Can we use `<EmailIcon />` here? That looks more appropriate imo.

File: new_client/src/components/auth/NewPw.jsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            label="New Password"
+            placeholder="Enter New Password"
+            type="password"
+            id="password"
+            autoComplete="current-password"
+            InputProps = {{startAdornment: <InputAdornment position="start"><VpnKeyIcon/></InputAdornment>}}
+            style={{color:'#52b107'}}
+            onChange={handlePassword}
+          />
+
```
Comment (ShubhankarKG) [inline new_client/src/components/auth/NewPw.jsx:123]: We are missing a div here. If there are errors, they need to be shown by a `<div>`

File: new_client/src/components/auth/ResetPw.jsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+          />
+
```
Comment (ShubhankarKG) [inline new_client/src/components/auth/ResetPw.jsx:112]: We are missing a div here. If there are errors, they need to be shown by a `<div>`

File: new_client/src/components/auth/Signin.jsx [hunk 7]
```diff
@@ -69,61 +122,89 @@ function Signin(props) {
...
+            fullWidth
+            color='#52b107'
+            required
+            label="Username"
+            name="username"
+            style={{color:'#52b107',borderColor:'#52b107'}}
+            onChange={handleUsername}
+      autoFocus
+      InputProps = {{startAdornment: <InputAdornment position="start"><AccountCircle/></InputAdornment>}}
+          />
```
Comment (ShubhankarKG) [inline new_client/src/components/auth/Signin.jsx:151]: We are missing a div here. If there are errors, they need to be shown by a `<div>`

File: new_client/src/components/auth/Signin.jsx [hunk 8]
```diff
@@ -69,61 +122,89 @@ function Signin(props) {
...
+            name="password"
+            label="Password"
+            type="password"
+            id="password"
+            autoComplete="current-password"
+            InputProps = {{startAdornment: <InputAdornment position="start"><VpnKeyIcon/></InputAdornment>}}
+            style={{color:'#52b107'}}
+            onChange={handlePassword}
+          />
+
```
Comment (ShubhankarKG) [inline new_client/src/components/auth/Signin.jsx:165]: We are missing a div here. If there are errors, they need to be shown by a `<div>`

**Review by ShubhankarKG — CHANGES_REQUESTED** [review]
Nice UI! Thanks for this, please fix the inline comments, we will then start with the backend part

File: new_client/src/components/auth/NewPw.jsx (hunk 5 above)
Comment (shekokar) [reply new_client/src/components/auth/NewPw.jsx]: Changes made.

**Comment by shekokar** [comment]
Have made the requested changes.

**Review by ShubhankarKG — COMMENTED** [review]
Thanks for the updates! The UI part looks solid to me, great work

**Comment by Saif807380** [comment]
The entire forget password flow works correctly! Just a few small changes -
- We can remove the `*` as the Email ID field is the only field on the page and thus is required by default
Screenshot from 2021-01-04 18-51-16
- Can we change the beginning of the email to 
```
Dear Full Name,

We have received ...
```
- Add a little more context to the subject of the email.
- Add a `Confirm New Password` Field when creating the new password.
- Add spinner after pressing update password.

**Comment by Saif807380** [comment]
Another small enhancement I forgot to mention. Can we implement a show/hide password within the password input?

**Comment by shekokar** [comment]
> The entire forget password flow works correctly! Just a few small changes -
> 
> * We can remove the `*` as the Email ID field is the only field on the page and thus is required by default
>   Screenshot from 2021-01-04 18-51-16
> * Can we change the beginning of the email to
> 
> ```
> Dear Full Name,
> 
> We have received ...
> ```
> 
> * Add a little more context to the subject of the email.
> * Add a `Confirm New Password` Field when creating the new password.
> * Add spinner after pressing update password.

Sure, will make the changes.

**Comment by shekokar** [comment]
> Another small enhancement I forgot to mention. Can we implement a show/hide password within the password input?

Will update this too

**Review by ShubhankarKG — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**